Create ui-c8y-1021-22-120-decode-application-options-passed-via-queryparameters-correctly-10138.md (#3827)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1021-22-120-decode-application-options-passed-via-queryparameters-correctly-10138.md

---
date: ""
title: decode application options passed via queryparameters correctly (#10138)
product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-64930
version: 1021.22.120
---

> Co-authored-by: Tristan Bastian <tristan.bastian@cumulocity.com>

Reviewer: ```suggestion
Previously, application options which where provided via query parameters to web applications, were only URI decoded in case the parameter contained valid JSON.
Now we decode these query parameters even when they are not containing valid JSON.
```

---
date: ""
title: Application options passed via query parameters are now decoded correctly
product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-64930
version: 1021.22.120
---

> Previously, application options which where provided via query parameters to web applications, were only URI decoded in case the parameter contained valid JSON.

Reviewer 2: ```suggestion
Previously, application options that were provided via query parameters to web applications were only URI-decoded if the parameter contained valid JSON.
```

> Now we decode these query parameters even when they are not containing valid JSON.

Reviewer 2: ```suggestion
With this change, these query parameters are also decoded if they do not contain valid JSON.
```
